list2/ex31/index.js

```diff
@@ -0,0 +1,49 @@
+import { generateMatrix, printMatrix } from "../helper.js";
+import promptSync from "prompt-sync";
+
+const prompt = promptSync();
+
+function comparingNumber({ matrix, columnCount, rowCount, comparativeNumber }) {
+  let equalComaparativeNumber = 0;
+  const matrixFiltered = matrix;
```

Comment:
That way when you change `matrixFiltered` you are also changing the original `matrix`. Please, make sure not to mutate parameters. 

![image](https://github.com/user-attachments/assets/bebab925-35b2-4351-988a-92092071ab9e)

Reply:
Done the right way

Reply:
Your code is still problematic. You are spreading only the first level of the matrix and this means that if you change any inner array, it will still mutate the original thing. If you want we can discuss it further since it is a very important topic.

![image](https://github.com/user-attachments/assets/11f209ce-62fb-4d85-9b7f-7dcd837d207d)